### PR TITLE
fix(cli) properly look for serf in known paths

### LINKED ISSUE #1953: kong fails to start due to missing serf
### Summary
I deployed a brand new Centos7 machine and fully updated it.

I installed Kong per installation guide (https://getkong.org/install/centos/), using this rpm: https://github.com/Mashape/kong/releases/download/0.9.7/kong-0.9.7.el7.noarch.rpm

I copied /etc/kong/kong.conf.defaut to /etc/kong/kong.conf and updated my database information.

However, kong fails to start:

Error:
/usr/local/share/lua/5.1/kong/cmd/start.lua:37: /usr/local/share/lua/5.1/kong/cmd/start.lua:22: could not find 'serf' executable (is 'serf_path' correctly set?)

Two solutions worked:

1: add to kong.conf: serf_path = /usr/local/bin/serf
2: add /usr/local/bin to root path

Simply put, following the installation instructions on a default CentOS 7 install will result in kong not starting due to serf not being found.

### Steps To Reproduce

1.
2.
3.
4.

### Additional Details & Logs

- Kong version (`$ kong version`)
- Kong debug-level startup logs (`$ kong start --vv`)
- Kong error logs (`<KONG_PREFIX>/logs/error.log`)
- Kong configuration (registered APIs/Plugins & configuration file)
- Operating System

### PR DESCRIPTION
**Pull request by thefosk**
### Summary

Our distributions put the `serf` executable in a known path at `/usr/local/bin/serf`. Adds this path as a known path (like we do for `resty`) when searching for Serf, if `serf_path` is not specified in the configuration file.

### Full changelog

* Look for `serf` in known paths, if `serf_path` is not specified.

### Issues resolved

Fix #1953

### CODE REVIEW

File: kong/cmd/utils/serf_signals.lua [hunk 1]
```diff
@@ -16,23 +16,46 @@ local serf_event_name = "kong"
...
     end
     return true
   end
+end
+
+local function check_serf_bin(kong_config)
+  log.debug("searching for 'serf' executable")
+
+  if kong_config.serf_path then
+    serf_search_paths = { kong_config.serf_path }
```
Comment (thibaultcha) [inline kong/cmd/utils/serf_signals.lua]: This shouldn't override all of the other potential locations. Maybe it should be replaced with:
```lua
table.insert(serf_search_paths, 1, kong_config.serf_path)
```

Comment (thefosk) [inline kong/cmd/utils/serf_signals.lua]: I thought about it - if I explicitly tell Kong to look into that path and it's not found, shouldn't an error be returned even if it's found somewhere else?

Does the `serf_path` property mean:

* search into `serf_path` in addition to the standard paths?
* look explicitly into `serf_path` and if it doesn't exist, return an error?

Seems like (2) would be a more expected behavior?

Comment (thibaultcha) [inline kong/cmd/utils/serf_signals.lua]: (1) would be a more expected behavior, similar to the shell's `$PATH` but also to the Lua's `LUA_PATH` and `LUA_CPATH` properties, also configurable from the Kong configuration.

Comment (thibaultcha) [inline kong/cmd/utils/serf_signals.lua]: It's also less error prone.

Comment (Tieske) [inline kong/cmd/utils/serf_signals.lua]: in other cases we said that if the user specifies a value, then not being able to honor that value is an error. We've had this discussion before.

If I would specify an argument "xyz" and the application starts and ends up running with "123" instead, that would be bad imo.

so it ought to be the nr 2. My 2cts.

Comment (thibaultcha) [inline kong/cmd/utils/serf_signals.lua]: This goes against all the other path variables in the shell and this project -_-

**Pull request merged**